src/charm.py

```diff
@@ -78,6 +110,171 @@ def configure_magma_access_gateway(orc8r_domain: str, root_ca_path: str) -> None
             stdout=subprocess.PIPE,
         )
 
+    def _is_configuration_valid(self) -> bool:
+        """Validates configuration."""
+        if self.model.config["skip-networking"]:
+            return True
+        valid = self._is_valid_interface("sgi", "eth0")
+        if not self._is_valid_interface("s1", "eth1"):
+            valid = False
+        if not self._is_valid_interface_addressing_configuration("sgi"):
+            valid = False
+        if not self._are_valid_dns(self.model.config["dns"]):
+            logger.warning("Invalid DNS configuration")
+            valid = False
+        return valid
+
+    def _is_valid_interface(self, interface_name: str, new_interface_name: str) -> bool:
+        """Validates a network interface name."""
```

Comment:
I think the docstring could be better. It's not obvious why we need the interface_name and a new_interface_name

Reply:
I updated the docstring, please let me know if it is clear enough.